pkg/bootstrapper/bootstrapper.go

```diff
@@ -876,17 +891,23 @@ func (cni *cniOptions) configure(kubeletCmd *string) error {
 // updateKubeletDependents updates the dependents field of the kubeletService struct
 // to reflect current list of dependent services. This function assumes that the kubelet service is running
 func updateKubeletDependents(svcMgr *mgr.Mgr) ([]*mgr.Service, error) {
-	var dependents []*mgr.Service
-	// If there is already a kubelet service running, find it
-	dependentSvc, err := svcMgr.OpenService(kubeletDependentSvc)
-	if err != nil {
-		// Do not return error if the services are not installed.
-		if !strings.Contains(err.Error(), "service does not exist") {
-			return nil, fmt.Errorf("error getting dependent services for kubelet %v", err)
+	// existingDependents is a list of dependent services that exist on the node
+	var existingDependents []*mgr.Service
+	// expectedDependents is a list of dependent services that could exist
+	// on the node, loop through this list to find the existing list of dependents
+	expectedDependents := []string{hybridOverlayDependentSvc, kubeletDependentSvc}
+	// find the existing service dependents
```

Comment:
For one, this sort of refactor should be a separate commit. It is also not clear why you are considering the transitive dependency, `hybridOverlayDependentSvc`, a direct dependent of the kubelet service.

Reply:
Doing this in simplest form possible here, I need to stop kube-proxy-> hybrid-overlay -> kubelet to stop kubelet service due to the dependency order. I can add comments to clarify this and add a seperate commit.

Reply:
I feel that the WMCB should return an error and not try to figure out the dependency graph. It should only be aware of the kubelet service. If it cannot stop that then it should return an error. It is up to the caller (WMCO)  to ensure that kube-proxy and hybrid-overlay are stopped before making this call.

Reply:
@aravindhp that makes sense, I ll drop handling of dependencies from here.